File: CoreTestDataConverter/Converter/StatusConverter.cs
using CoreTestDataConverter.Model;
using SaveData.Model;
using SaveData.Model.Development;

namespace CoreTestDataConverter.Converter
{
	public static class StatusConverter
	{
		public static StatusTestData ToTestData(StatusSaveData saveData)
		{
			if (saveData == null)
			{
				return null;
			}

			return new StatusTestData
			{
				Attack = saveData.Attack,
				HP = saveData.HP,
				Defense = saveData.Defense,
				Speed = saveData.Speed,
				CritRate = saveData.CritRate,
				CritDamage = saveData.CritDamage,
				EffectHit = saveData.EffectHit,
				EffectResist = saveData.EffectResist
			};
		}

		public static FullStatusTestData ToTestData(FullStatusSaveData saveData)
		{
			if (saveData == null)
			{
				return null;
			}

			return new FullStatusTestData
			{
				Attack = saveData.Attack,
				HP = saveData.HP,
				Defense = saveData.Defense,
				Speed = saveData.Speed,
				CritRate = saveData.CritRate,
				CritDamage = saveData.CritDamage,
				EffectHit = saveData.EffectHit,
				EffectResist = saveData.EffectResist,
				AdditionalAttackRate = saveData.AdditionalAttackRate,
				AdditionalHpRate = saveData.AdditionalHpRate,
				AdditionalDefenseRate = saveData.AdditionalDefenseRate,
			};
		}
	}
}

File: CoreTest/TestData/Converter/StatusConverter.cpp
#include "pch.h"
#include "StatusConverter.h"

StatusTestData StatusConverter::toTestData(const nlohmann::json& json)
{
	StatusTestData data{};

	data.Attack = json.at("Attack").get<double>();
	data.HP = json.at("HP").get<double>();
	data.Defense = json.at("Defense").get<double>();
	data.Speed = json.at("Speed").get<double>();
	data.CriticalRate = json.at("CritRate").get<double>();
	data.CriticalDamage = json.at("CritDamage").get<double>();
	data.EffectHit = json.at("EffectHit").get<double>();
	data.EffectResist = json.at("EffectResist").get<double>();

	return data;
}

Status StatusConverter::toNative(const StatusTestData& testData)
{
	Status status{};

	status.Attack = testData.Attack;
	status.Hp = testData.HP;
	status.Defense = testData.Defense;
	status.Speed = testData.Speed;
	status.CriticalRate = testData.CriticalRate;
	status.CriticalDamage = testData.CriticalDamage;
	status.EffectHit = testData.EffectHit;
	status.EffectResist = testData.EffectResist;

	return status;
}
